src/backend/dx11/src/factory.rs

```diff
@@ -589,7 +589,7 @@ impl core::Factory<R> for Factory {
             }
             layouts.push(d3d11::D3D11_INPUT_ELEMENT_DESC {
                 SemanticName: &charbuf[charpos],
-                SemanticIndex: 0,
+                SemanticIndex: u32::from(attrib.slot),
```

Comment:
why isn't this just `attrib.slot as u32` or something like this?

Reply:
Clippy has yelled at me in the past about this: https://rust-lang.github.io/rust-clippy/master/#cast_lossless. Though I'm happy to change it to `as u32` if that's preferred.